lib/winappdriver.js

```diff
@@ -130,8 +132,38 @@ class WinAppDriver extends events.EventEmitter {
   }
 
   async startSession (caps) {
+
+    const createSessionTimeout = caps.createSessionTimeout || this.createSessionTimeout;
+    log.debug(`Start WinAppDriver session during createSessionTimeout = '${createSessionTimeout}' ms.`);
     this.proxyReqRes = this.jwproxy.proxyReqRes.bind(this.jwproxy);
-    await this.jwproxy.command('/session', 'POST', {desiredCapabilities: caps});
+    let retryIteration = 0;
+    let lastError;
+
+    const condFn = async () => {
+      lastError = null;
+      try {
+        retryIteration++;
+        await this.jwproxy.command('/session', 'POST', {desiredCapabilities: caps});
+        return true;
+      } catch (error) {
+        lastError = error;
+        log.warn(`Could not start WinAppDriver session error = '${error}', attempt = '${retryIteration}' from '${this.createSessionRetry}'`);
+        return false;
+      }
+    };
+
+    try {
+      await waitForCondition(condFn, {
+        waitMs: createSessionTimeout,
+        intervalMs: 500
+      });
+    } catch (timeoutError) {
+      log.debug(`timeoutError was ${timeoutError}`);
+      if (lastError) {
+        throw (lastError);
+      }
+      throw new Error(`Could not start WinAppDriver session during timeout '${createSessionTimeout}' ms.`);
```

Comment:
Could not start WinAppDriver session within '${createSessionTimeout}' ms.

Reply:
Done

Reply:
BTW with the current implementation of waitForCondition this Error will be never thrown as it does not check success result on timing out.